tests/python_functional/functional_tests/destination_drivers/snmp_destination/001-snmp-destination-v2c-normal-working-compared-cisco-ios-traps-default-values.py

```diff
@@ -0,0 +1,83 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import os
+import time
+
+import pytest
+
+from src.helpers.snmptrapd.snmptrapd import SNMPtrapd
+
+
+CISCO_TIMETICKS = '97881'
+CISCO_FACILITY = "SYS"
+CISCO_SEVERITY = "6"
+CISCO_MESSAGE_TYPE = "CONFIG_I"
+CISCO_MESSAGE = "Configured from console by vty1 (10.30.0.32)"
+
+input_log = "<38>Feb 11 21:27:22 testhost testprogram[9999]: test message\n"
+expected_log = 'iso.3.6.1.4.1.9.9.41.1.2.3.1.2.55 = STRING: "' + CISCO_FACILITY + '"	iso.3.6.1.4.1.9.9.41.1.2.3.1.3.55 = INTEGER: ' + CISCO_SEVERITY + '	iso.3.6.1.4.1.9.9.41.1.2.3.1.4.55 = STRING: "' + CISCO_MESSAGE_TYPE + '"	iso.3.6.1.4.1.9.9.41.1.2.3.1.5.55 = STRING: "' + CISCO_MESSAGE + '"	iso.3.6.1.4.1.9.9.41.1.2.3.1.6.55 = Timeticks: (' + CISCO_TIMETICKS + ')'
+
+
+@pytest.mark.skipif(os.system("which snmptrapd") != 0, reason="snmptrapd can't found on system")
+@pytest.mark.parametrize(
+    "input_log, expected_log, counter", [
+        (input_log, expected_log, 1),
+    ], ids=["with_one_log"],
+)
+def test_acceptance(config, syslog_ng, input_log, expected_log, counter):
+    # create source driver config
+    file_source = config.create_file_source(file_name="input.log")
+
+    # create destination driver config
+    snmp_objs = (
+        "'1.3.6.1.4.1.9.9.41.1.2.3.1.2.55', 'Octetstring', '%s'" % CISCO_FACILITY,
+        "'1.3.6.1.4.1.9.9.41.1.2.3.1.3.55', 'Integer', '%s'" % CISCO_SEVERITY,
+        "'1.3.6.1.4.1.9.9.41.1.2.3.1.4.55', 'Octetstring', '%s'" % CISCO_MESSAGE_TYPE,
+        "'1.3.6.1.4.1.9.9.41.1.2.3.1.5.55', 'Octetstring', '%s'" % CISCO_MESSAGE,
+        "'1.3.6.1.4.1.9.9.41.1.2.3.1.6.55', 'Timeticks', '%s'" % CISCO_TIMETICKS,
+    )
+
+    trap_obj = "'.1.3.6.1.6.3.1.1.4.1.0','Objectid','.1.3.6.1.4.1.9.9.41.2.0.1'"
+
+    snmp_nonstandard_port = 2222
+    snmptrapd = SNMPtrapd(snmp_nonstandard_port)
+    snmptrapd.start()
+
+    snmp_destination = config.create_snmp_destination(
+        host="'127.0.0.1'",
+        port=snmp_nonstandard_port,
+        version="'v2c'",
+        snmp_obj=snmp_objs,
+        trap_obj=trap_obj,
+    )
+
+    config.create_logpath(statements=[file_source, snmp_destination])
+    config.update_global_options(keep_hostname="yes")
+    file_source.write_log(input_log, counter)
+
+    syslog_ng.start(config)
+
+    time.sleep(1)
```

Comment:
Is this sleep necessary? What is the purpose?

Reply:
no, it isn't. removed